gcloud/datastore/test_query.py

```diff
@@ -33,9 +24,17 @@ def _getTargetClass(self):
     def _makeOne(self, *args, **kw):
         return self._getTargetClass()(*args, **kw)
 
-    def test_ctor_defaults(self):
-        query = self._getTargetClass()()
-        self.assertEqual(query.dataset, None)
+    def test_ctor_defaults_wo_implicit_dataset_id(self):
+        _DATASET = 'DATASET'
+        self.assertRaises(ValueError, self._makeOne)
+
+    def test_ctor_defaults_w_implicit_dataset_id(self):
+        from gcloud._testing import _Monkey
+        from gcloud.datastore import _implicit_environ
+        _DATASET = 'DATASET'
+        with _Monkey(_implicit_environ, DATASET_ID=_DATASET):
```

Comment:
Thanks for doing this. Much preferred to the maintenance of `setUp` and `tearDown`.